syndeo-template/src/test/java/com/google/cloud/syndeo/transforms/SyndeoPubsubReadSchemaTransformProviderTest.java

```diff
@@ -0,0 +1,159 @@
+
+  private static final Schema BEAMSCHEMA =
+      Schema.of(
+          Schema.Field.of("name", Schema.FieldType.STRING),
+          Schema.Field.of("number", Schema.FieldType.INT64));
+  private static final String SCHEMA = AvroUtils.toAvroSchema(BEAMSCHEMA).toString();
+  private static final String SUBSCRIPTION = "projects/project/subscriptions/subscription";
+  private static final String TOPIC = "projects/project/topics/topic";
+
+  private static final List<Row> ROWS =
+      Arrays.asList(
+          Row.withSchema(BEAMSCHEMA)
+              .withFieldValue("name", "a")
+              .withFieldValue("number", 100L)
+              .build(),
+          Row.withSchema(BEAMSCHEMA)
+              .withFieldValue("name", "b")
+              .withFieldValue("number", 200L)
+              .build(),
+          Row.withSchema(BEAMSCHEMA)
+              .withFieldValue("name", "c")
+              .withFieldValue("number", 300L)
+              .build());
+
+  private static final Clock CLOCK = (Clock & Serializable) () -> 1656788475425L;
+
+  private static final AvroPayloadSerializerProvider AVRO_PAYLOAD_SERIALIZER_PROVIDER =
+      new AvroPayloadSerializerProvider();
+  private static final PayloadSerializer AVRO_PAYLOAD_SERIALIZER =
+      AVRO_PAYLOAD_SERIALIZER_PROVIDER.getSerializer(BEAMSCHEMA, new HashMap<>());
+
+  @Rule public TestPipeline p = TestPipeline.create().enableAbandonedNodeEnforcement(false);
+
+  @Test
+  public void testInvalidInput() {
+    PCollectionRowTuple begin = PCollectionRowTuple.of("BadInput", p.apply(Create.of(ROWS)));
+    assertThrows(
+        IllegalArgumentException.class,
+        () ->
+            begin.apply(
+                new SyndeoPubsubReadSchemaTransformProvider()
+                    .from(
+                        SyndeoPubsubReadSchemaTransformConfiguration.builder()
+                            .setSchema(SCHEMA)
+                            .setFormat("AVRO")
+                            .build())
+                    .buildTransform()));
+  }
+
+  @Test
+  public void testReadAvro() throws IOException {
+    PCollectionRowTuple begin = PCollectionRowTuple.empty(p);
+
+    try (PubsubTestClientFactory clientFactory =
+        clientFactory(incomingAvroMessagesOf(CLOCK.currentTimeMillis()))) {
+      SyndeoPubsubReadSchemaTransformConfiguration config =
+          SyndeoPubsubReadSchemaTransformConfiguration.builder()
+              .setFormat("AVRO")
+              .setSchema(SCHEMA)
+              .setTopic(TOPIC)
+              .setClientFactory(clientFactory)
+              .build();
+      SchemaTransform transform = new SyndeoPubsubReadSchemaTransformProvider().from(config);
+      PCollectionRowTuple reads = begin.apply(transform.buildTransform());
+
+      PAssert.that(reads.get("output")).containsInAnyOrder(ROWS);
+
+      p.run().waitUntilFinish();
+      clientFactory.close();
+    } catch (Exception e) {
+      throw new IllegalArgumentException(e.toString());
```

Comment:
```suggestion
      throw new IllegalArgumentException(e.toString(), e);
```

you need to pass the original exception to get the proper source for it